File: src/wwwroot/app/common/models/business/ownership/ownership.ts
/// <reference path="../../../../typings/_all.d.ts" />

module Antares.Common.Models.Business {
    export class Ownership implements Dto.IOwnership {
        id: string = null;
        createDate: Date = null;
        purchaseDate: Date = null;
        sellDate: Date = null;
        buyPrice: number = null;
        sellPrice: number;
        isCurrentOwner: boolean;
        ownershipTypeId: string;
        ownershipType: Dto.IOwnershipType;
        contacts: Contact[];

        constructor(ownership?: Dto.IOwnership) {
            if (ownership) {
                angular.extend(this, ownership);
                this.contacts = ownership.contacts.map((value: Dto.IContact) => { return new Contact(value); });
                this.isCurrentOwner = !this.sellDate;
            }
        }

        public isVendor(): boolean {
            return this.isCurrentOwner && this.ownershipType.code == Enums.OwnershipTypeEnum.Freeholder;
        }
    }
}

File: src/KnightFrank.Antares.Search/Models/Ownership.cs
namespace KnightFrank.Antares.Search.Models
{
    using System;
    using System.Collections.Generic;

    public class Ownership
    {
        public string Id { get; set; }

        public DateTime? PurchaseDate { get; set; }

        public DateTime? SellDate { get; set; }

        public string PropertyId { get; set; }

        public decimal? BuyPrice { get; set; }

        public decimal? SellPrice { get; set; }

        public string OwnershipTypeId { get; set; }

        public EnumTypeItem OwnershipType { get; set; }

        public IList<OwnershipContact> OwnershipContacts { get; set; }
    }
}
